### ISSUE TITLE
no function clause matching in :io.request/2

### ISSUE THREAD

**Issue by houshuang** [mneudert/plug_accesslog#8]
Got this error in my log file

``` elixir
2015-07-12 17:46:54.724 [error] GenEvent handler Plug.AccessLog.Writer installed in Plug.AccessLog.Writer terminating
Last message: {:log, "4 [12/Jul/2015:17:46:54 +0800] GET \"/resource/list\" 200 25994 25932", "log/access_log"}
State: #PID<0.707.0>
** (exit) an exception was raised:
    ** (FunctionClauseError) no function clause matching in :io.request/2
        (stdlib) io.erl:555: :io.request({nil, nil}, {:put_chars, :unicode, "4 [12/Jul/2015:17:46:54 +0800] GET \"/resource/list\" 200 25994 25932\n"})
        (stdlib) io.erl:62: :io.o_request/3
        (plug_accesslog) lib/plug/accesslog/writer.ex:23: Plug.AccessLog.Writer.handle_event/2
        (elixir) lib/gen_event.ex:990: GenEvent.do_handler/3
        (elixir) lib/gen_event.ex:831: GenEvent.server_update/5
        (elixir) lib/gen_event.ex:816: GenEvent.server_notify/7
        (elixir) lib/gen_event.ex:789: GenEvent.server_event/4
        (elixir) lib/gen_event.ex:578: GenEvent.handle_msg/5
```

**Comment by mneudert** [comment]
Looks to me like there was a problem opening the logfile. Could that have happened due to logrotate or something else manipulating that file?

I will add some logging to the areas accessing the logfiles. That should at least give some more insights (and strike of one more point on the todo list ^^).

The logging is still up and running after that error or did something completely shut off the event handling?

**Comment by houshuang** [comment]
This was locally while developing, and there is no log_rotate in place. It
could have been confused by Phoenix reloading or something, but I don't
think this was immediately after a reload. The request did not complete.

On Sun, Jul 12, 2015 at 7:15 PM, Marc Neudert notifications@github.com
wrote:

> Closed #8 https://github.com/mneudert/plug_accesslog/issues/8 via
> cf6cbbc
> https://github.com/mneudert/plug_accesslog/commit/cf6cbbceef2730add11d53114377e52297dddf9b
> .
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/mneudert/plug_accesslog/issues/8#event-353757128.

## 

http://reganmian.net/blog -- Random Stuff that Matters

**Comment by mneudert** [comment]
That one request definitely should have killed the logging process while NOT preventing the web request from completing. My changes should catch all cases where a file could not be opened because of whatever reasons. The reasons should be written to Logger.error/1 while just skipping the log entry for that request (and all following while that problem exists...).
